src/core/algorithms/cind/condition.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <boost/container_hash/hash.hpp>
+
+#include "condition_miners/itemset_node.h"
+#include "condition_miners/position_lists_set.h"
+#include "table/encoded_column_data.h"
+
+namespace algos::cind {
+char const* const kAnyValue = "-";
+
+struct Condition {
+    std::vector<std::string> condition_attrs_values;
+    double validity;
+    double completeness;
+
+    Condition(std::shared_ptr<ItemsetNode> itemset,
+              std::vector<model::EncodedColumnData const*> const& condition_attrs)
+        : validity(itemset->GetValidity()), completeness(itemset->GetCompleteness()) {
+        condition_attrs_values.resize(condition_attrs.size(), kAnyValue);
+        ItemsetNode const* item_ptr = itemset.get();
+        for (size_t column_id = condition_attrs.size() - 1; column_id >= 0; --column_id) {
```

Comment:
`column_id >= 0` is always true, because `size_t` is unsigned type. It doesn't seem to matter in what order we view them, so it's easier to use a direct traversal.
```suggestion
        for (size_t column_id = 0; column_id < condition_attrs.size() ; ++column_id) {
```